include/circt/Dialect/SV/TypeDecl.td

```diff
@@ -116,6 +116,35 @@ def InterfaceModportOp : SVOp<"interface.modport",
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// Types for integration in to type-based dialects.
+//===----------------------------------------------------------------------===//
+
+def Interface : SVType<"Interface"> {
+  let summary = "SystemVerilog interface type pointing to an InterfaceOp";
+  let description = [{
+    A MLIR type for the SV dialect's `InterfaceOp` to allow instances in any
+    dialect with an open type system. Points at the InterfaceOp which defines
+    the SystemVerilog interface.
+  }];
+
+  let mnemonic = "interface";
+  let parameters = (ins "::mlir::SymbolRefAttr":$interface);
+}
+
+
+def InterfaceModport : SVType<"InterfaceModport"> {
```

Comment:
Does SV have other things called "modport's"?  If not, I'd recommend naming this ModportType and change the syntax to be `!sv.modport..` instead of `!sv.interface.modport...`